controllers/ibmpowervscluster_controller.go

```diff
@@ -111,6 +113,114 @@ func (r *IBMPowerVSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	return r.reconcile(clusterScope)
 }
 
+type powerVSCluster struct {
+	cluster *infrav1beta2.IBMPowerVSCluster
+	mu      sync.Mutex
+}
+
+type reconcileResult struct {
+	reconcile.Result
+	error
+}
+
+func (update *powerVSCluster) updateCondition(condition bool, conditionArgs ...interface{}) {
+	update.mu.Lock()
+	defer update.mu.Unlock()
+	if condition {
+		conditions.MarkTrue(update.cluster, conditionArgs[0].(capiv1beta1.ConditionType))
+		return
+	}
+
+	conditions.MarkFalse(update.cluster, conditionArgs[0].(capiv1beta1.ConditionType), conditionArgs[1].(string), conditionArgs[2].(capiv1beta1.ConditionSeverity), conditionArgs[3].(string), conditionArgs[4:]...)
+}
+
+func (r *IBMPowerVSClusterReconciler) reconcilePowerVSResources(clusterScope *scope.PowerVSClusterScope, powerVSCluster *powerVSCluster, ch chan reconcileResult, wg *sync.WaitGroup) {
+	defer wg.Done()
+	powerVSLog := clusterScope.WithName("powervs")
+	// reconcile PowerVS service instance
+	powerVSLog.Info("Reconciling PowerVS service instance")
+	if requeue, err := clusterScope.ReconcilePowerVSServiceInstance(); err != nil {
+		powerVSLog.Error(err, "failed to reconcile PowerVS service instance")
+		powerVSCluster.updateCondition(false, infrav1beta2.ServiceInstanceReadyCondition, infrav1beta2.ServiceInstanceReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
+		ch <- reconcileResult{reconcile.Result{}, err}
+		return
+	} else if requeue {
+		powerVSLog.Info("PowerVS service instance creation is pending, requeuing")
+		ch <- reconcileResult{reconcile.Result{Requeue: true}, nil}
+		return
+	}
+	powerVSCluster.updateCondition(true, infrav1beta2.ServiceInstanceReadyCondition)
+
+	clusterScope.IBMPowerVSClient.WithClients(powervs.ServiceOptions{CloudInstanceID: clusterScope.GetServiceInstanceID()})
+
+	// reconcile network
+	powerVSLog.Info("Reconciling network")
+	if requeue, err := clusterScope.ReconcileNetwork(); err != nil {
+		powerVSLog.Error(err, "failed to reconcile PowerVS network")
+		powerVSCluster.updateCondition(false, infrav1beta2.NetworkReadyCondition, infrav1beta2.NetworkReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
+		ch <- reconcileResult{reconcile.Result{}, err}
+		return
+	} else if requeue {
```

Comment:
we aren't doing any `requeue` here, hence it will be good if we can rename or rewrite the logic to have meaningful name here

Reply:
also please add why there is no requeue here?

Reply:
I have refactored the code, please take a look!